### PR TITLE
FMS 2023.04 change and using relative paths in config

### LINKED ISSUE #340: (SE) A design choice: Using absolute vs. relative paths
While trying to run tier1 tests with the `3dvar_atmos` suite, I received the following error:

```bash  
FATAL from PE     0: get_ascii_file_num_lines: File /discover/nobackup/gmao_ci/swell/tier1/9003594169/3dvar_atmos/swell-3dvar_atmos-9003594169/stage/fv3-jedi/geos_atmosphere/fv3fil does not exist.

Image              PC                Routine            Line        Source
libifcoremt.so.5   00002B2AA45C928B  tracebackqq_          Unknown  Unknown
libfms.so          00002B2A9608E51E  mpp_mod_mp_mpp_er     Unknown  Unknown
```

I noticed the length of the missing path is 128 characters and that is likely the culprit. I've had this issue with `SOCA` (MOM6) as well, it's a limitation of the `fms` handling .nml files. After talking to @rtodling he mentioned some newer `fms` versions do support 256/512 characters, however there is some discrepancy between JCSDA and GEOS versions in terms of `fms`, perhaps ours is newer, @mathomp4?

While designing the Swell tasks, the idea was to be able to execute them from any location, hence we started using absolute paths. However, it's also possible to just use `os.chdir` command inside the scripts to ensure we are in the proper directory instead of using absolute paths (similar to how it's done in slurm *.sh scripts).

I'm not sure if there are any reasons to avoid relative paths but I'm happy to hear any opposing thoughts?

### PR DESCRIPTION
**Pull request by Dooruk**
## Purpose
1) In recent JEDI builds, FMS is loaded as a module rather than building it as an external package. Hence, this PR handles the minor `modules` modifications. The bulk of the work is in the `jedi_bundle` wrapper (see https://github.com/GEOS-ESM/jedi_bundle/pull/41 and https://github.com/GEOS-ESM/jedi_bundle/pull/46)

2) FMS has a limitation of allowing up to 128 characters in path length. There will be a future version allowing up to 1024 characters, but we need to stick with relative paths in the interim.

## Impact
`StageJedi` task had two modes:

 1) (without datetime input) Linking static files to the `stage` directory in the experiment folder. This has the advantage of running this task once.
 2) (with datetime input) Linking files to `cycle_dir` at each cycle. This has the advantage of allowing relative paths.
 
 This PR skips the first option to surmount path length issue.

@rtodling, this points to a recent JEDI build from July 24th `/discover/nobackup/projects/gmao/advda/swell/JediBundles/fv3_soca_SLES12/`. While we figure out the pinned versions transition you can run and compare your previous variational outputs with this one, if they compare we can move ahead with the pinned version changes.

Closes https://github.com/GEOS-ESM/swell/issues/340

### CODE REVIEW

**Comment by rtodling** [comment]
I suppose this should work even for older versions of JEDI - w/o the FMS update ... I will give this a try in pointing to my sendbox tomorrow.

**Comment by rtodling** [comment]
I just tried this out and it is failing in GetBackgroundGeosExper... and in GetObservations

**Comment by Dooruk** [comment]
> I just tried this out and it is failing in GetBackgroundGeosExper... and in GetObservations

Sorry, I just rebuild it again, needed a SOCA fix. Can you give it a shot again?

Testing Tier 1 with this branch, you can watch it live here:

https://github.com/GEOS-ESM/swell/actions/runs/10164687694

**Comment by rtodling** [comment]
Let me try again ... since I had an different r2d2 yaml file ... trying w/ the default now ...

Still get the same error:

  File "/gpfsm/dnb05/projects/p61/rtodling/JEDI1/Feb21/opt/core/swell/Mar5/lib/python3.10/site-packages/swell/tasks/base/task_base.py", line 287, in task_wrapper
    task_object.execute()

  File "/gpfsm/dnb05/projects/p61/rtodling/JEDI1/Feb21/opt/core/swell/Mar5/lib/python3.10/site-packages/swell/tasks/get_background_geos_experiment.py", line 85, in execute
    self.geos.linker(bkgr_tar, bkgr_tar_file, dst_dir=self.cycle_dir())

  File "/gpfsm/dnb05/projects/p61/rtodling/JEDI1/Feb21/opt/core/swell/Mar5/lib/python3.10/site-packages/swell/utilities/geos.py", line 183, in linker
    self.logger.abort(f'Source file does not exist: {src}')

in GetBackgroundGeosExp...

**Comment by rtodling** [comment]
/gpfsm/dnb05/projects/p139/rtodling/cylc-run/swell-3dfgat_atmos-suite/run1/log/job/20211212T0000Z/GetBackgroundGeosExperiment-geos_atmosphere/01

**Comment by Dooruk** [comment]
> /gpfsm/dnb05/projects/p139/rtodling/cylc-run/swell-3dfgat_atmos-suite/run1/log/job/20211212T0000Z/GetBackgroundGeosExperiment-geos_atmosphere/01

That task will require for you to provide X experiment backgrounds via `geos_x_background_directory`, it is seeing the `/dev/null` default, which doesn't exist. That shouldn't fail the experiment though, you can see Tier 1 ran.

**Comment by Dooruk** [comment]
@rtodling, I tested `3dvar_atmos` on Github with the following. Tier 1 test now can run `3dvar_atmos` with the relative path change: 

https://github.com/GEOS-ESM/swell/actions/runs/10165110908

**Comment by rtodling** [comment]
> @rtodling, I tested `3dvar_atmos` on Github with the following. Tier 1 test now can run `3dvar_atmos` with the relative path change:
> 
> https://github.com/GEOS-ESM/swell/actions/runs/10165110908

Ok - you are right about that ... the run finished successfully  after I let it run ...


Actually - very sorry - the run does not complete successfully. There is an error in CRTM connection:

simobs(FAILURE) : Error allocating CRTM K-matrix Surface structure (setTraj)


I am trying this again ... I think I had an slurm.py sting in my sandbox that messed things up ...

**Review by rtodling — APPROVED** [review]
After making a couple of mistakes of my own doing I can not confirm that this works fine.

Thank you so much.
